src/main/java/cz/sparko/boxitory/service/filesystem/FilesystemDigestHashService.java

```diff
@@ -74,8 +74,23 @@ public String getHashType() {
 
     @Override
     public String getChecksum(String box) {
-        final String hash = hashStore.loadHash(box, hashAlgorithm)
-                .orElseGet(() -> calculateHash(box));
+        return hashStore.loadHash(box, hashAlgorithm)
+                .orElseGet(() -> calculateHashAndStore(box));
+    }
+
+    @Override
+    public String getChecksum(String box, boolean canCalculate) {
```

Comment:
can you please write unit test for this method and refactor it? It can be simplified and duplicate code with `getChecksum` can be removed by simple call of that method

Reply:
Done